app/models/vmdb_database/seeding.rb

```diff
@@ -38,19 +49,63 @@ def db_disk_size(disk)
       return nil if err.message.include?("does not exist")
       raise
     end
-  end
 
-  def seed
-    mine = evm_tables.includes(:text_tables, :vmdb_indexes).index_by(&:name)
+    def seed_tables
+      db = my_database
+      vmdb_tables = db.vmdb_tables.index_by(&:name)
+
+      table_names = connection.tables
+      tables = table_names.zip([]).to_h
+      tables.merge!(connection.text_table_names.slice(*table_names))
+
+      tables.each do |t, tt|
+        table = vmdb_tables.delete(t)
+        table ||= VmdbTableEvm.create!(:name => t, :vmdb_database => db) do
+          _log.info("Creating VmdbTableEvm #{t.inspect}")
+        end
+
+        next unless tt
+
+        text_table = vmdb_tables.delete(tt)
+        text_table ||= VmdbTableText.new
 
-    self.class.connection.tables.sort.each do |table_name|
-      table = mine.delete(table_name)
-      VmdbTableEvm.seed_for_database(self, table || table_name)
+        text_table.attributes = {:name => tt, :evm_table => table, :vmdb_database => db}
+        if text_table.changed?
+          _log.info("#{text_table.new_record? ? "Creating" : "Updating"} VmdbTableText #{tt.inspect} for VmdbTableEvm #{t.inspect}")
+        end
+        text_table.save!
+      end
+
+      if vmdb_tables.any?
+        _log.info("Deleting the following VmdbTable(s) as they no longer exist: #{vmdb_tables.keys.sort.collect(&:inspect).join(", ")}")
+        VmdbTable.delete(vmdb_tables.values.map(&:id))
+      end
     end
 
-    mine.each do |name, t|
-      _log.info("Table <#{name}> is no longer in Database <#{self.name}> - deleting")
-      t.destroy
+    def seed_indexes
+      db = my_database
+      vmdb_tables  = db.vmdb_tables.index_by(&:name)
+      vmdb_indexes = db.vmdb_indexes.index_by(&:name)
+
+      indexes = connection.index_names.slice(*vmdb_tables.keys)
```

Comment:
Seems like you could decrease things on this line (and the two above it), but just pulling back the `name` and `id` for each query, and use `:vmdb_table_id` down when setting the attributes instead.

Also, I don't know where `connection.index_names` is coming from... I tried this in the console with no dice:

```ruby
irb> ActiveRecord::Base.connection.index_names
```

Reply:
[connection.index_names](https://github.com/ManageIQ/manageiq/pull/18331/files#diff-066d248b63363e78cb77b2232e584b54R493)

Reply:
@NickLaMuro Marking this as resolved because I did this on the querying line

Reply:
> [connection.index_names](https://github.com/ManageIQ/manageiq/pull/18331/files#diff-066d248b63363e78cb77b2232e584b54R493)

Derp... sorry... I even tried looking for that... in the file...